### PR TITLE
Security Analysis: Fix Generator initialTargetP in OperatorStrategies

### PR DESCRIPTION
**Pull request by jeandemanged**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**

Bug fix

**What is the current behavior?**

If there is a single operator strategy for a contingency, the operator strategy simulation is done using initialTargetP of the basecase state instead of the post-contingency state.
Note: No issue if there are multiple operator strategies for a continency, in this case all operator strategies simulations are done using initialTargetP of the post-contingency state, as it should be. But there is a bug in the case of a single unique operator strategy for the contingency.

**What is the new behavior (if this is a feature change)?**
Operator strategies correctly start from the good starting point always.

**Does this PR introduce a breaking change or deprecate an API?**
- [x] No

**Other information**:

Fix of https://github.com/powsybl/powsybl-open-loadflow/pull/1041

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/sa/AbstractSecurityAnalysis.java [hunk 1]
```diff
@@ -456,8 +456,11 @@ protected SecurityAnalysisResult runSimulations(LfNetwork lfNetwork, List<Propag
 
                                 List<OperatorStrategy> operatorStrategiesForThisContingency = operatorStrategiesByContingencyId.get(lfContingency.getId());
                                 if (operatorStrategiesForThisContingency != null) {
-                                    // we have at least an operator strategy for this contingency.
+                                    // we have at least one operator strategy for this contingency.
                                     if (operatorStrategiesForThisContingency.size() == 1) {
+                                        // only one operator strategy, no need to do a complete save of network state,
+                                        // but need to set generators initialTargetP positions to the current (=postContingency) targetP
+                                        lfNetwork.setGeneratorsInitialTargetPToTargetP();
```
Comment (jeandemanged) [inline src/main/java/com/powsybl/openloadflow/sa/AbstractSecurityAnalysis.java:463]: note to reviewers: this only line is the fix

**Review by vmouradian — APPROVED** [review]

**Review by annetill — APPROVED** [review]
Approved, but maybe these feature should be documented for a better long-term support.

File: docs/security/security.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+## Active power slack distribution and Operator Strategies Active Power relative actions
+When slack distribution is enabled in the load flow options, then slack distribution is performed in all network 
+simulations of the security analysis, that is:
+- in the N ("pre-contingency") simulation, resulting in the creation of the pre-contingency state
+- in the N-1 or N-k ("post-contingency") simulation, performed by applying contingencies on the pre-contingency state,
+resulting in the creation of the post-contingency state
+- in the Operator Strategies ("curative") simulation, performed by applying actions on the post-contingency state,
+resulting in the creation of the curative state
+
+Actions defined in Operators Strategies and involving active power changes (e.g. Load Actions, Generator Actions) can be defined as relative.
```
Comment (annetill) [inline docs/security/security.md:14]: Thanks for your doc, it is a very nice beginning. Just a remark: I thought you will talk about initial targetP management... no need?

Comment (jeandemanged) [reply docs/security/security.md:14]: added an example, hopefully would be clearer

File: docs/security/parameters.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+## Generic parameters
+
+Generic parameters are documented in [PowSyBl Core](inv:powsyblcore:*:*#simulation/security/configuration)
+
+The next section details the parameters that are specific to PowSyBl Open LoadFLow.
+
+## Specific parameters
+
+**contingencyPropagation**  
+The `contingencyPropagation` property:
```
Comment (annetill) [inline docs/security/parameters.md:18]: Maybe we can mention that contingency propagation is only relevant for node/breaker topology, no?

File: docs/security/parameters.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+- when set to `false`, Security Analysis will simulate the outage of only the equipment(s) defined in the contingency.
+- when set to `true` (default) Security Analysis will determine by topological search the switches with type circuit breakers
+(i.e. capable of opening fault currents) that must be opened to isolate the fault. Depending on the network structure,
+this could lead to more equipments to be simulated as tripped, because disconnectors and load break switches
+(i.e., not capable of opening fault currents) are not considered.
+
+The default value is `true`.
+
+**createResultExtension**  
+The `createResultExtension` property defines whether open-loadflow specific results extensions should be created
```
Comment (annetill) [inline docs/security/parameters.md:28]: open-loadflow, Open LoadFlow, what do you choose ?

File: docs/security/parameters.md (hunk 3 above)
Comment (annetill) [reply docs/security/parameters.md:21]: In iidm topology is defined at the level of the voltage level, meaning that even if most of the time, all a network is of same topology but hybrid network is supported.

File: docs/security/parameters.md (hunk 4 above)
Comment (jeandemanged) [reply docs/security/parameters.md]: neither, I choose Open Load Flow :)

File: docs/security/parameters.md (hunk 3 above)
Comment (jeandemanged) [reply docs/security/parameters.md]: improved this part

**Review by annetill — APPROVED** [review]
Nice!

**Pull request merged**